### PR TITLE
docs: add code-overview from wiki to src/nvim/README.md

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by clason** [comment]
Maybe move this to the "Nvim Lifecycle" section (removing duplicate information, if any?

**Comment by dundargoc** [comment]
Don't think there's much duplicate info tbh

File: src/nvim/README.md [hunk 1]
```diff
@@ -392,6 +392,68 @@ modes managed by the `state_enter` loop:
...
+
+The main loop is implemented in state_enter. The basic idea is that Vim waits
+for the user to type a character and processes it until another character is
+needed.  Thus there are several places where Vim waits for a character to be
+typed.  The `vgetc()` function is used for this.  It also handles mapping.
+
+Updating the screen is mostly postponed until a command or a sequence of
+commands has finished.  The work is done by `update_screen()`, which calls
+`win_update()` for every window, which calls `win_line()` for every line.
+See the start of [drawscreen.c](../blob/master/src/nvim/drawscreen.c) for more explanations.
```
Comment (zeertzjq) [inline src/nvim/README.md:416]:
```suggestion
See the start of [drawscreen.c](drawscreen.c) for more explanations.
```

File: src/nvim/README.md [hunk 2]
```diff
@@ -392,6 +392,68 @@ modes managed by the `state_enter` loop:
...
+### Ex commands
+
+Ex commands are handled by the function `do_cmdline()`.  It does the generic
+parsing of the `:` command line and calls `do_one_cmd()` for each separate
+command.  It also takes care of while loops.
+
+`do_one_cmd()` parses the range and generic arguments and puts them in the
+exarg_t and passes it to the function that handles the command.
+
+The `:` commands are listed in [ex_cmds.lua](../blob/master/src/nvim/ex_cmds.lua).
```
Comment (zeertzjq) [inline src/nvim/README.md:434]:
```suggestion
The `:` commands are listed in [ex_cmds.lua](ex_cmds.lua).
```

File: src/nvim/README.md [hunk 3]
```diff
@@ -392,6 +392,68 @@ modes managed by the `state_enter` loop:
...
+
+### Normal mode commands
+
+The Normal mode commands are handled by the `normal_cmd()` function.  It also
+handles the optional count and an extra character for some commands.  These
+are passed in a `cmdarg_T` to the function that handles the command.
+
+There is a table `nv_cmds` in [normal.c](../blob/master/src/nvim/normal.c) which
```
Comment (zeertzjq) [inline src/nvim/README.md:442]:
```suggestion
There is a table `nv_cmds` in [normal.c](normal.c) which
```

File: src/nvim/README.md [hunk 4]
```diff
@@ -392,6 +392,68 @@ modes managed by the `state_enter` loop:
...
+
+### Insert mode commands
+
+When doing an `i` or `a` command, `normal_cmd()` will call the `edit()` function.
+It contains a loop that waits for the next character and handles it.  It
+returns when leaving Insert mode.
+
+### Options
+
+There is a list with all option names in [options.lua](../blob/master/src/nvim/options.lua).
```
Comment (zeertzjq) [inline src/nvim/README.md:455]:
```suggestion
There is a list with all option names in [options.lua](options.lua).
```

**Comment by dundargoc** [comment]
Right. I'm still hesitant to touch it, I don't want to mess anything up since i'm not familiar with it.

**Review by justinmk — APPROVED** [review]
Great that we are eliminating the wiki. We probably can delete this wiki page since it isn't targeting by many external links.

**Review by clason — APPROVED** [review]

**Pull request merged**